r/tests/testthat/test-embedding.R

```diff
@@ -0,0 +1,37 @@
+mock_req <- function() {
+  req <- list(
+    body = list(
+      type = "umap",
+      config = list(minimumDistance = 0.1, distanceMetric = "cosine")
+    )
+  )
+}
+
+mock_scdata <- function() {
+  data("pbmc_small", package = "SeuratObject", envir = environment())
+  pbmc_small$cells_id <- 0:(ncol(pbmc_small) - 1)
+  pbmc_small@misc$gene_annotations <- data.frame(
+    input = paste0("ENSG", seq_len(nrow(pbmc_small))),
+    name = row.names(pbmc_small),
+    row.names = paste0("ENSG", seq_len(nrow(pbmc_small)))
+  )
```

Comment:
I see that this `mock_scdata` is used in all test functions. You can extract it in a different place and then import and reuse it in individual tests to reduce the copy-paste

Reply:
also, you might want to consider adding now or in the future other types of datasets for testing with specific characteristics to test for wider variety of cases.

Reply:
Each dataset is different depending on the set of tests, that's why we have a different one per file. I'm happy to chat about changing this if you think it should be changed

Reply:
I didn't realise this, makes sense! Then leave it as is